Spatial.gd

```diff
@@ -621,31 +629,49 @@ func gen_new_icon(texture_path: String, waypoint_icon: Waypoint.Icon, category_i
 	var category_data = category_item.get_metadata(0)
 	category_data.category3d.add_icon(new_icon)
 
-# This function take all of the currently rendered objects and converts it into
-# the data format that is saved/loaded from.
-func data_from_renderview():
-	var icons_data = []
-	var paths_data = []
-
-	for icon in $Icons.get_children():
-		icons_data.append({
-			"position": [icon.translation.x, icon.translation.y, -icon.translation.z],
-			"texture": icon.texture_path
-		})
-
-	for path in $Paths.get_children():
-		#print(path)
-		var points = []
-		for point in range(path.get_point_count()):
-			var point_position:Vector3 = path.get_point_position(point)
-			points.append([point_position.x, point_position.y, -point_position.z])
-		paths_data.append({
-			"points": points,
-			"texture": path.texture_path
-		})
-
-	var data_out = {"icons": icons_data, "paths": paths_data}
-	return data_out
+################################################################################
+# Section of functions for saving data to file
+################################################################################
+func save_map_data(map_id: int):
+	var packed_bytes = self.waypoint_data.to_bytes()
+	var file = File.new()
+	file.open(self.marker_file_path, file.WRITE)
+	file.store_buffer(packed_bytes)
+
+func get_hash(data: PoolByteArray) -> String:
+	var ctx = HashingContext.new()
+	ctx.start(HashingContext.HASH_MD5)
+	ctx.update(data)
+	var res: PoolByteArray = ctx.finish()
+	return res.hex_encode()
+
+const FILE_HASH_PATH: String = "user://file_hash.json"
+
+# Save all hashes
+func save_hashes():
+	var file = File.new()
+	var data = {}
+	var dir = Directory.new()
+	dir.open(self.unsaved_markers_dir)
+	dir.list_dir_begin()
+	var file_name = dir.get_next()
+	while file_name != "":
+		if dir.file_exists(file_name):
```

Comment:
This seems like a weird check:
"for each file in the directory if the file exists do a thing"

Reply:
The while loop here also returns the name of subdirectories. The other option here was to use `if not dir.current_is_dir()` as there is not `current_is_file` or such.

Reply:
I see, I think it would be better to use something like `is_file` unless `file_exists` is the only way to do that.

Reply:
There is no `is_file` function from what I can find.

Reply:
Then the way you have it is probably the right thing to do, the godot syntax is just weird. Not surprising given how weird the Directory/File syntax is.